rmf_robot_sim_common/src/slotcar_common.cpp

```diff
@@ -86,6 +86,84 @@ double compute_friction_energy(
   return f * m * g * v * dt;
 }
 
+// Given a line segment from point1 to point2, and a circle centred at (cx, cy),
+// returns the points of intersection.
+std::vector<Eigen::Vector2d> line_circle_intersections(
```

Comment:
Nit, I would suggest making the functions `static` since they are local to this file